### PR TITLE
[Function builders] Run syntactic diagnostics for function bodies.

### PR DESCRIPTION
**Pull request by DougGregor**
When a function body has had a function builder applied to it, make
sure that we run all of the syntactic diagnostics for expressions and
statements within the body.

Fixes rdar://problem/64493626.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please test source compatibility

File: lib/Sema/BuilderTransform.cpp [hunk 1]
```diff
@@ -1387,6 +1387,63 @@ BraceStmt *swift::applyFunctionBuilderTransform(
         captured.first, captured.second)));
 }
 
+/// Produce any additional syntactic diagnostics for the body of a
```
Comment (theblixguy) [inline lib/Sema/BuilderTransform.cpp:1390]: Looks like this doc comment is incomplete :)

Comment (DougGregor) [reply lib/Sema/BuilderTransform.cpp:1390]: Fixed locally, thanks! I'm going to let CI run a bit before pushing the change.

**Pull request merged**